sycl/include/sycl/handler.hpp

```diff
@@ -2870,6 +2893,25 @@ class __SYCL_EXPORT handler {
   void memcpyFromDeviceGlobal(void *Dest, const void *DeviceGlobalPtr,
                               bool IsDeviceImageScoped, size_t NumBytes,
                               size_t Offset);
+
+  template <typename T, int Dims, access::mode AccessMode,
+            access::target AccessTarget,
+            access::placeholder IsPlaceholder = access::placeholder::false_t,
+            typename PropertyListT = property_list>
+  void checkIfPlaceholderIsBoundToHandler(
+      accessor<T, Dims, AccessMode, AccessTarget, IsPlaceholder, PropertyListT>
+          Acc) {
+    detail::AccessorBaseHost *AccBase = (detail::AccessorBaseHost *)&Acc;
+    detail::AccessorImplPtr AccImpl = detail::getSyclObjImpl(*AccBase);
+    detail::AccessorImplHost *Req = AccImpl.get();
+    detail::ArgDesc AD(detail::kernel_param_kind_t::kind_accessor, Req,
+                       static_cast<int>(AccessTarget), 0);
+    if (std::find(MAssociatedAccesors.begin(), MAssociatedAccesors.end(), AD) ==
+        MAssociatedAccesors.end())
```

Comment:
Could we have a case where the entry we are looking for doesn't have `MIndex > 0`? I suspect we can. I think we better remove the new equality operator and just make the check in here instead, e.g.

```suggestion
    auto FindRequirementArgDesc = [&](const ArgDesc &AD) {
      return AD.MType == detail::kernel_param_kind_t::kind_accessor &&
                 AD.MPtr == Req &&
                  // Do we need this last check?
                  // Can we have different access targets with the same requirement?
                 AD.MSize == static_cast<int>(AccessTarget);
    };
    if (std::find_if(MAssociatedAccesors.begin(), MAssociatedAccesors.end(), FindRequirementArgDesc) ==
        MAssociatedAccesors.end())
```

Reply:
[28e20bd](https://github.com/intel/llvm/pull/8531/commits/28e20bda7edbbabe8d56d02cbaa946c94fbad6f4)
I'm not sure about `AD.MSize == static_cast<int>(AccessTarget);`, so I suggest to keep this check.

Reply:
Thank you! Since `MIndex` has no bearing on it, I figure it's best to keep it out then. 😄